tests/classmethods_test.py

```diff
@@ -22,12 +22,25 @@
 from mockito import when, unstub, verify
 from mockito.verification import VerificationError
 
-class Dog:
+
+class LoudBarker:
+    @classmethod
+    def loud_bark(cls, bark):
+        return bark
+
+
+class Dog(LoudBarker):
```

Comment:
I don't like it if we change the `Dog` structure and meaning for the old tests.  Esp. if you look below when we explicitly do a old-school `Lion(object)` it looks like inheritance could matter.  So I think it would be better to define two classes just for this purpose.  

To test the wanted behavior, it is also important that we check that we actually don't change or mock the base class `LoudBarker` when mocking the `Barker`. Basically in pseudo-code

```
when(Barker).loud_bark....
assert LoudBarker.loud_bark(...)  # ensure not mocked
```

And that we get the old classes back after unstubbing. That's really important for inheritance mocking because usually we shadow the original method when mocked, and need to *delete* the shadowing classmethod on unstub. In contrast: replacing it with the original method would not be the same because then a mocked base `Barker` would probably not change the behavior of the `LoudBarker` anymore.

Reply:
Agreed.  I'll rework this.